### PR TITLE
Add example that reports diagnostics

### PR DESCRIPTION
**Pull request by chsienki**

### CODE REVIEW

**Comment by chsienki** [comment]
@dotnet/roslyn-compiler for review please.

File: docs/features/source-generators.cookbook.md [hunk 1]
```diff
@@ -212,6 +212,58 @@ public partial class {userClass.Identifier}
...
+For code-based issues, the generator author should also consider implementing a [diagnostic analyzer](https://docs.microsoft.com/en-us/visualstudio/code-quality/roslyn-analyzers-overview?view=vs-2019) that identifies the problem, and offers a code-fix to resolve it.
+
+**Example:**
+
+```csharp
+[Generator]
+public class MyXmlGenerator : ISourceGenerator
+{
+
+    private static readonly DiagnosticDescriptor InvalidXmlWarning = new DiagnosticDescriptor(id: "MYXMLGEN001",
```
Comment (jcouv) [inline docs/features/source-generators.cookbook.md:231]: >MYXMLGEN [](start = 99, length = 8)

just curious, are some diagnostic prefixes disallowed as we do in analyzers in general? (for example "CS")

Comment (chsienki) [reply docs/features/source-generators.cookbook.md:231]: As far as I know, we don't actually disallow prefixes, but we do have an analyzer that warns it shouldn't be used. I've confirmed that the analyzer issues the warnings for the generator in the above case.

@mavasani to confirm the expected behavior.

Comment (mavasani) [reply docs/features/source-generators.cookbook.md:231]: Yep, see https://github.com/dotnet/roslyn-analyzers/issues/1727 and https://github.com/dotnet/roslyn/issues/25748 for more context. We attempted to restrict prefixes in the compiler, but that caused a breaking change, so we reverted it and instead implemented an analyzer that warns when using well-known/reserved prefixes such as CS, BC or CA.

**Comment by RikkiGibson** [comment]
For expediency I am not including this in the preview2 snap this afternoon. Please let me know if you have concerns.

**Pull request merged**